src/status_im/contexts/profile/edit/name/events.cljs

```diff
@@ -0,0 +1,14 @@
+(ns status-im.contexts.profile.edit.name.events
+  (:require [utils.i18n :as i18n]
+            [utils.re-frame :as re-frame]))
+
+(re-frame/reg-event-fx :profile/edit-name
+ (fn [{:keys [db]} [name]]
+   {:db            (update db :profile/profile merge {:display-name name})
```

Comment:
You are always overwriting `display-name` on profile, so it's better to use `assoc-in`.

```clojure
(assoc-in db [:profile/profile :display-name] name)
```